lib/auth/webauthncli/fido2.go

```diff
@@ -702,6 +702,15 @@ func withRetries(callback deviceCallbackFunc) deviceCallbackFunc {
 			// constants.
 			var fidoErr libfido2.Error
 			if !errors.As(err, &fidoErr) {
+				// Handle here errors mapped by go-libfido2.
+				switch {
+				// ErrOperationDenied is returned when wrong finger is used on bio reader.
+				case errors.Is(err, libfido2.ErrOperationDenied):
+					fmt.Println("Gesture validation failed, make sure you use register fingerprint")
+					log.Debug("FIDO2: Retrying libfido2 error 'operation denied'")
+					continue
+				}
+
 				return err
 			}
```

Comment:
Suggestion: cut nesting and remove Println? We don't print anything for other retries.

(I'm using an if because there's only one branch, but once we add more we can swap it for a switch.)

```suggestion
			// Handle errors mapped by go-libfido2.
			// ErrOperationDenied happens when fingerprint reading fails (UV=false).
			if errors.Is(err, libfido2.ErrOperationDenied) {
				log.Debug("FIDO2: Retrying libfido2 error 'operation denied'")
				continue
			}

			// Handle generic libfido2.Error instances.
			var fidoErr libfido2.Error
			if !errors.As(err, &fidoErr) {
				return err
			}
```

Reply:
@codingllama if someone use wrong finger, how will he know that he/she should "rethink" if finger used is correct? that was my idea behind println